go/engine/deprovision_test.go

```diff
@@ -342,3 +415,32 @@ func TestCurrentDeviceRevoked(t *testing.T) {
 	tc.G.SecretStoreAll = nil
 	assertCurrentDeviceRevoked(tc)
 }
+
+func TestDeprovisionLastDevice(t *testing.T) {
+	testDeprovisionLastDevice(t, false)
+}
+
+func TestDeprovisionLastDevicePUK(t *testing.T) {
+	testDeprovisionLastDevice(t, true)
+}
+
+// A user should be able to revoke all of their devices.
+func testDeprovisionLastDevice(t *testing.T, upgradePerUserKey bool) {
+	tc := SetupEngineTest(t, "deprovision")
+	defer tc.Cleanup()
+	tc.Tp.UpgradePerUserKey = upgradePerUserKey
+	if tc.G.SecretStoreAll == nil {
+		t.Fatal("Need a secret store for this test")
+	}
+	fu := assertDeprovisionWithSetup(tc, assertDeprovisionWithSetupArg{
+		revokePaperKey: true,
+	})
+	assertNumDevicesAndKeys(tc, fu, 0, 0)
+
+	// Now, test deprovision codepath with no secret store
+	tc.G.SecretStoreAll = nil
+	fu = assertDeprovisionWithSetup(tc, assertDeprovisionWithSetupArg{
+		revokePaperKey: true,
+	})
+	assertNumDevicesAndKeys(tc, fu, 0, 0)
+}
```

Comment:
Yes, there is `TestRevokerPaperDeviceTwicePUK` which does this:
- signup (adds puk generation 1)
- make a second paper key
- revoke paper key (adds puk generation 2)
- revoke paper key (adds puk generation 3)
- checks the keyring before and after a sync to make it sure it is on generation 3. It doesn't check the actual key contents.